src/tests/e2e/login.post.test.ts

```diff
@@ -0,0 +1,88 @@
+import 'dotenv/config';
+import request from 'supertest';
+import app from '../../app';
+import { mockPassword, mockUsername, mockBasicAuthString } from '../mockObjects/mockUser';
+import { setupE2E } from '../setupTests';
+import { ReasonPhrases, StatusCodes } from 'http-status-codes';
+import { userService } from '../../services/userService';
+import { mockComparePassword, mockGenerateTokens, mockGetUser } from '../mockObjects/mockUserService';
+
+const ENDPOINT = '/login';
+
+setupE2E('login-e2e', () => {
+  const generateTokensSpy = jest.spyOn(userService, 'generateTokens').mockImplementation(mockGenerateTokens);
+  const getUsersSpy = jest.spyOn(userService, 'getUser').mockImplementation(mockGetUser);
+  const comparePasswordSpy = jest.spyOn(userService, 'comparePassword').mockImplementation(mockComparePassword);
+  describe('POST /login responds', () => {
+    test('with set of tokens for a logged in user', async () => {
+      const res = await request(app)
+        .post(ENDPOINT)
+
+        .set('Authorization', mockBasicAuthString(mockUsername, mockPassword));
+      expect(res.statusCode).toBe(StatusCodes.OK);
+      expect(res.body.accessToken).toBeDefined();
+      expect(res.body.refreshToken).toBeDefined();
+    });
+
+    test(`with ${StatusCodes.UNAUTHORIZED} when username is not provided`, async () => {
```

Comment:
That's a test for "when username is too short/empty". It would be not provided if you passed null